### ISSUE TITLE
Database dump contains no data when using "gitea dump -d postgres"

### ISSUE THREAD

**Issue by mpraeger** [go-gitea/gitea#5659]
- Gitea version (or commit ref): 1.6.3, 17.0-rc2
- Git version: 2.20.1
- Operating system: FreeBSD 11.2
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description
I'm trying to migrate from MySQL to PostgreSQL and was advised (see #5651) that using *gitea dump -d postgres* might work. While this does indeed produce an SQL dump the inserts do not contain any data (see [dump.txt](https://github.com/go-gitea/gitea/files/2734118/dump.txt) ).

I've used the following command:
```
gitea dump -c /usr/local/etc/gitea/conf/app.ini -d postgres
```
If there's anything missing or further information is required I'd be more than happy to provide it.

Thanks!

**Comment by typeless** [comment]
Maybe you can try to add '-v' to get more details.

Edit: Looking at the log message and https://github.com/go-gitea/gitea/blob/master/cmd/dump.go might give some clues.

**Comment by mpraeger** [comment]
Absolutely: [gitea_dump_verbose.txt](https://github.com/go-gitea/gitea/files/2736348/gitea_dump_verbose.txt)

Unfortunately gitea does not seem to be logging much for the database dump.

**Comment by typeless** [comment]
@mpraeger Do you have the log file `xorm.log` in the generated zip file?

**Comment by mpraeger** [comment]
@typeless Had to create a new dump, but sure: [xorm.log](https://github.com/go-gitea/gitea/files/2737299/xorm.txt)

**Comment by mpraeger** [comment]
I've just tested dumping to sqlite3, that produced a proper dump.

**Comment by mpraeger** [comment]
Alright so going from MySQL to sqlite3 and then issuing `gitea dump -d postgres` produces a proper dump.

**Comment by gzsombor** [comment]
I accidentally found the same issue, and figured out, how to fix : https://github.com/go-xorm/xorm/pull/1186

**Comment by lunny** [comment]
@gzsombor merged in that xorm's PR, please send a vendor update PR on gitea.

**Comment by techknowlogick** [comment]
Closed with #5680
